hooks/pre-push

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+set -e
+
+echo '+cargo test'
```

Comment:
Is there a way to only run this if at least one .rs file was touched since the last push?

Reply:
Yes sure. Will do tomorrow.

Reply:
Unfortunately not, I have not been able to run `clippy` on specific files.

Reply:
My proposed solution here is to get the file names from the diff with master and only run cargo test when there are rust changes and only run pytest when there are python changes. Keeping in mind that this is no "smart-script" so the branch diff check will always be in comparison with master. Also not that the `pre-push` hook is not meant to be necessary (only intended to make the dev's life easier).

Reply:
Check out the new and improved pre-push!

```
➜  dex-services git:(hooks) git push
Skipping cargo test; Branch does not differ from master in a rusty way!
Skipping pytest; Branch does not differ from master in a pythonic way!
PUSH SUCCEEDED
```